### ISSUE TITLE
2nd-level domains

### ISSUE THREAD

**Issue by KorvinSzanto** [dot-src/dot-src#5]
I believe that domains should be limited to official programming languages, and we should market subdomains.

{LANG}.src can list popular / trending projects for easy identification, projects can reserve {PROJECT}.{LANG}.src with something like a pull request, given that every bit of code is publicly accessible in an open source format with a license they can choose from a list of acceptable license.

Project uri's would look something like:
- jquery.js.src
- backbone.js.src
- flask.py.src
- php.php.src

etc.

**Comment by joallard** [comment]
Some early thoughts. It's a good idea.

The reserve I have with 2nd level domains (2LDs) is that project names are usually unique across languages, and it only adds to verbosity. 

Take Backbone, for example. The community has organized so that "backbone" is a unique name across not only JS, but Ruby, Python, PHP, etc. So if we're putting it under the JS 2LD, we're basically saying "that name could exist under another 2LD." I don't think that's our aim here. I think we have to coordinate one code namespace (for projects) consisting of unique names across languages.

I do like the take on file extensions though. Domains like `py.src` would obviously point to the right language and a directory is a good idea.

**Comment by nirix** [comment]
What if there _are_ projects with the same name though, would we then register them as `{project}.{lang}.src` or give the `.src` to the first project to register it even though the other may be more active with a larger community?

**Comment by joallard** [comment]
That would make sense.

**Comment by KorvinSzanto** [comment]
I think those are the sort of things you deal with after you've procured the gtld for 300k

**Comment by joallard** [comment]
We're never too prepared!

**Comment by cedricdonie** [comment]
I'm for keeping things such as `*.py.src`. Otherwise, we would end up with something like `nodejs.src` instead of `node.js.src`. (IMO the second one is better)

**Comment by mhelvens** [comment]
> I believe that domains should be limited to official programming languages

If you don't mind me asking: why? I'm not sure what an _unofficial_ programming language is, but if I wrote open source code in one, why shouldn't I be allowed to host it under a `.src` domain?

> and we should market subdomains

I'm not a fan of this idea. Allow me to present my reasoning.

First of all, a great number of projects use multiple languages. Many, if not most, 'JavaScript projects' also contain HTML and CSS. A couple of mine also have Typescript and Sass thrown in. And while I personally use Node.js, many web-stacks do not use the same language for client and server.

And even if you could identify a 'main language', this is only one of the properties by which one could categorize open source projects, and not necessarily the most interesting. What about: library vs app, target-audience, platform, goal, and so on.

Certainly, the (main) programming language is a useful piece of information, but not a logical one to put in the URL. What if a project switches from one PL to another? Would they have to get a new domain-name?

That said, there are certainly projects that are tightly linked to a specific language. Many libraries with 'js' in their name come to mind. By all means, set up `.js.src` for aesthetic purposes.

PS: The creation of a searchable index of hosted projects is certainly a good idea (and one should certainly be able to filter projects by PL). Realize that this would require time and money to set up and maintain. But it might be a good selling point for potential investors.

**Comment by cedricdonie** [comment]
@mhelvens Hmm you're correct it seems.
How about we offer both? third level domains would then just be cheaper, as we can sell more of those. I would, depending on the project like to host it under a third-level domain, or a second level. I say we register the 2/3 letter codes for the second level domains, but otherwise offer freedom of choice.

**Comment by joallard** [comment]
Yep, that's what I was thinking. Reserve the short codes for 2nd level, other wise leave it open. In Node's case (which is particular since JS is already in the name), we could offer to reserve both `node.js.src` and `nodejs.src`, and have a redirect from one to the other. 

Basically, you could get a second-level domain unless your name is not unique across languages.

**Comment by arxpoetica** [comment]
Then again, sometimes I like distinguishing between `node` and `js`...`node.src` vs `js.src`?
